### PR TITLE
Milestone 8: Recipe App

### PR DESCRIPTION
**Pull request by shyusu4**
## PR Description
- Used database migration files to maintain database schema
- Used validations for models
- Secured app from `N+1` problems
- Wrote SQL queries with ActiveRecord
- Set up associations between models
- Built a web app that requires the user to log in
- Described the difference between `authorization` and `authentication`
- Use Devise gem for authentication
- Understood how `sessions` and `cookies` can support authentication
- Limited access to web app resources based on authorization rules

**_Please note that I have completed this project solo due to the way Microverse has sorted this week's groups._**

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                View
+              </a>
+              </div>
+            <div class="flex flex-col p-4 w-50">
+              <div><%= recipe.description %></div>
+            </div>
+          </div>
+        </div>
+        </div>
+      <% end %>
```
Comment (Omar-Muhamad) [inline app/views/recipes/index.html.erb:19]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=You%20should%20follow%20the%20layout%20of%20the%20wireframes%20provided.), Please stick to the given wireframe design and the placement of the elements and take care of the contrast ratio between font color and background color.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          <th  scope="col" class="px-6 py-3"> Value </th>
+          	<th scope="col" class="px-6 py-3"> Action </th>
+        </tr>
+      </thead>
+      <tbody>
+
+        <% RecipeFood.where(recipe_id: params[:id]).each do |recipe_food| %>
+					<tr  class="bg-white border-b dark:bg-gray-800 dark:border-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600">
+					<td  scope="row" class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap dark:text-white">
+						<%= recipe_food.foods.name %>
```
Comment (Omar-Muhamad) [inline app/views/recipes/show.html.erb:32]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=You%20should%20follow%20the%20layout%20of%20the%20wireframes%20provided.), Please center the element in the page to match the given wireframe design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image | image  |

File: app/views/recipe_foods/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  <%= form.label :quantity  %>
+  <%= form.number_field :quantity %>
+  </div>
+            <div>
+            <%= form.label :food_id %>
+  <%= form.select(:food_id, Food.all.map{|u| [u.name, u.id]},{ include_blank: true}) %></div>
+  <div>
+    <%= form.submit "Add Ingredient"%>
+    <%= link_to "Back to recipes", recipes_path%>
+  </div>
```
Comment (Omar-Muhamad) [inline app/views/recipe_foods/new.html.erb:14]: - [ ] There is an error raised when I try to add ingredients to a recipe, Please fix that.

image

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @shyusu4,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by uwadonat — APPROVED** [review]
## Hi @shyusu4,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good commit history ✔️
- Professional pull request message ✔️


209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196


# Status: Approved ✔️✔️✔️
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**